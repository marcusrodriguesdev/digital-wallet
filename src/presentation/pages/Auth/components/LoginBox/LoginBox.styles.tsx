import styled from 'styled-components'

export const Container = styled.div`
  display: flex;
  flex-direction: column;
`

export const Title = styled.h1`
  ${({ theme }) => theme?.fonts?.h3 ?? {}}
  color: ${({ theme }) => theme?.colors?.main ?? '#fffffe'};
  margin-bottom: 50px;
`

export const InputWrapper = styled.div`
  margin-bottom: 40px;
`

export const ButtonWrapper = styled.div``
