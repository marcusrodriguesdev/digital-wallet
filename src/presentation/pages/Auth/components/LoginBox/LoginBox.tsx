import React from 'react'

import { Button } from '@/presentation/components/Button'
import { Input } from '@/presentation/components/Input'

import { ButtonWrapper, Container, InputWrapper, Title } from './LoginBox.styles'

export interface LoginBoxProps {
  emailValue: string
  disabled: boolean
  onSubmit: () => void
  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void
}

const LoginBox: React.FC<LoginBoxProps> = (props: LoginBoxProps) => {
  const { emailValue, disabled, onSubmit, onChange } = props

  return (
    <Container>
      <Title>Welcome</Title>
      <InputWrapper>
        <Input label="E-mail" name="email" value={emailValue} type="text" scale="large" onChange={onChange} />
      </InputWrapper>
      <ButtonWrapper>
        <Button label="Next" scale="large" variant="default" disabled={disabled} onClick={onSubmit} />
      </ButtonWrapper>
    </Container>
  )
}

export { LoginBox }
