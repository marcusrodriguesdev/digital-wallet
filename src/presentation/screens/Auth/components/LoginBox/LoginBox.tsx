import React from 'react'
import { Button } from '@/presentation/components/Button'
import { Input } from '@/presentation/components/Input'
import { ButtonWrapper, Container, InputWrapper, Title } from './LoginBox.styles'

export interface LoginBoxProps {
  emailValue: string
  onSubmit: () => void
  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void
}

const LoginBox: React.FC<LoginBoxProps> = (props: LoginBoxProps) => {
  const { emailValue, onSubmit, onChange } = props

  return (
    <Container>
      <Title>Login</Title>
      <InputWrapper>
        <Input label="E-mail" name="email" value={emailValue} type="text" scale="large" onChange={onChange} />
      </InputWrapper>
      <ButtonWrapper>
        <Button label="Entrar" scale="large" variant="default" onClick={onSubmit} />
      </ButtonWrapper>
    </Container>
  )
}

export { LoginBox }
