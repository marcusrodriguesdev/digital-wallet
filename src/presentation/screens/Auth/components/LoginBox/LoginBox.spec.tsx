import React from 'react'
import { LoginBox, LoginBoxProps } from './LoginBox'

import * as Faker from 'faker'
import { fireEvent, render } from '@testing-library/react'

const makeSUT = (props?: Partial<LoginBoxProps>): any => {
  const emailValue = props?.emailValue ?? Faker.random.words(2)
  const onSubmitSpy = jest.fn()
  const onChangeSpy = jest.fn()

  const sut = <LoginBox emailValue={emailValue} onSubmit={onSubmitSpy} onChange={onChangeSpy} />

  return { sut, emailValue, onSubmitSpy, onChangeSpy }
}

describe('presentation/screens/Auth/LoginBox', () => {
  it('should match snapshot', () => {
    const { sut } = makeSUT({
      emailValue: 'email@email.com'
    })
    const loginBoxRender = render(sut)
    expect(loginBoxRender).toMatchSnapshot()
  })

  it('should call onSubmit with correct values', () => {
    const { sut, onSubmitSpy } = makeSUT()
    const { getByTestId } = render(sut)
    const button = getByTestId('Button/Button')
    button.click()
    expect(onSubmitSpy).toHaveBeenCalled()
  })

  it('should call onChange with correct values', () => {
    const { sut, onChangeSpy } = makeSUT()
    const { getByTestId } = render(sut)
    const input = getByTestId('Input/StyledInput')
    fireEvent.change(input, { target: { value: 'any_value' } })
    expect(onChangeSpy).toHaveBeenCalled()
  })
})
