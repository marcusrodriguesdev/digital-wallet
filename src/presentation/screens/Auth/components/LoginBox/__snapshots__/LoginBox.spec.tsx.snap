// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`presentation/screens/Auth/LoginBox should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-ftTHYK eDlwTg"
      >
        <h1
          class="sc-pyfCe csxPwX"
        >
          Login
        </h1>
        <div
          class="sc-jrcTuL khraKV"
        >
          <div
            class="sc-dkrFOg gZiSLJ"
          >
            <a
              class="sc-hLBbgP bjecrs"
              data-testid="Input/Label"
              scale="large"
            >
              E-mail
            </a>
            <input
              class="sc-eDvSVe hJjMXc"
              data-testid="Input/StyledInput"
              name="email"
              scale="large"
              type="text"
              value="email@email.com"
            />
          </div>
        </div>
        <div
          class="sc-kDvujY"
        >
          <button
            class="sc-bcXHqe eNsOjl"
            data-testid="Button/Button"
            scale="large"
            type="button"
          >
            <span
              class="sc-gswNZR eYthHy"
              scale="large"
            >
              Entrar
            </span>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-ftTHYK eDlwTg"
    >
      <h1
        class="sc-pyfCe csxPwX"
      >
        Login
      </h1>
      <div
        class="sc-jrcTuL khraKV"
      >
        <div
          class="sc-dkrFOg gZiSLJ"
        >
          <a
            class="sc-hLBbgP bjecrs"
            data-testid="Input/Label"
            scale="large"
          >
            E-mail
          </a>
          <input
            class="sc-eDvSVe hJjMXc"
            data-testid="Input/StyledInput"
            name="email"
            scale="large"
            type="text"
            value="email@email.com"
          />
        </div>
      </div>
      <div
        class="sc-kDvujY"
      >
        <button
          class="sc-bcXHqe eNsOjl"
          data-testid="Button/Button"
          scale="large"
          type="button"
        >
          <span
            class="sc-gswNZR eYthHy"
            scale="large"
          >
            Entrar
          </span>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
